build-utils/src/org/pentaho/wadl/PentahoResourceDoclet.java

```diff
@@ -0,0 +1,129 @@
+package org.pentaho.wadl;
+
+import java.io.BufferedOutputStream;
+import java.io.FileOutputStream;
+import java.io.OutputStream;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.Marshaller;
+
+import com.sun.jersey.wadl.resourcedoc.ResourceDoclet;
+import org.apache.xml.serialize.OutputFormat;
+import org.apache.xml.serialize.XMLSerializer;
+
+import com.sun.javadoc.AnnotationDesc;
+import com.sun.javadoc.ClassDoc;
+import com.sun.javadoc.MethodDoc;
+import com.sun.javadoc.RootDoc;
+import com.sun.jersey.server.wadl.generators.resourcedoc.model.ClassDocType;
+import com.sun.jersey.server.wadl.generators.resourcedoc.model.MethodDocType;
+import com.sun.jersey.server.wadl.generators.resourcedoc.model.ResourceDocType;
+import com.sun.jersey.wadl.resourcedoc.DocProcessorWrapper;
+
+public class PentahoResourceDoclet extends ResourceDoclet {
+  private static final String OUTPUT_FILE_NAME = "wadlExtension.xml";
+  private static final String OUTPUT_PATH_PARAM = "-output";
+
+  private static boolean isDeprecated( AnnotationDesc[] annotationDescs ) {
+    for ( AnnotationDesc annotationDesc : annotationDescs ) {
+      if ( "deprecated".equalsIgnoreCase( annotationDesc.annotationType().name() ) ) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  private static boolean isPublic( String documentation ) {
+    return documentation != null && !"".equals( documentation.trim() );
```

Comment:
Agree. We can search for that too. But we added it probably because
https://github.com/pentaho/pentaho-platform/blob/master/extensions/src/org/pentaho/platform/web/http/api/resources/SchedulerResource.java#L341
Do they represent the same?